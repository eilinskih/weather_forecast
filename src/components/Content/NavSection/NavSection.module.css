.AppWrapper { 
    background: black;
    height: 100%;
    border-radius: 25px;
    position: relative;
    left: -20px;
    padding-top: 50px;
    color: rgb(172, 172, 172);
}

.weatherInfo {
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.weatherInfo h4 {
    color: honeydew;
    font-size: 38px;
}
 
.icon {
    display: flex;
    flex-direction: column;
}

.icon span {
    font-size: 30px;
}

.foreCast {
    font-size: 18px;
    margin-top: 20px;
    display: flex;
    justify-content: space-around;
    background: green;
    width: 100%;
}

.day {
    display: flex;
    flex-direction: column;
}

.sunset {
    font-size: 18px;
    display: flex;
    justify-content: space-around;
    margin-top: 50px;
}

.group { 
    margin-top: 50px;
    display: flex;
    justify-content: space-between;
    padding: 0 30px;
}

.something {
    border-radius: 30px 30px 30px 0;
    background:rgb(42, 43, 36);
    padding: 10px;
}

.wind {
    border-radius: 30px 0 30px 30px;
    background:rgb(42, 43, 36);
    padding: 10px;
}

.AppWrapper h3 { 
    color: honeydew;
}