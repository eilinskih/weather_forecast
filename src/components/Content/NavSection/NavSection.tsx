import { CloudOutlined } from '@ant-design/icons';
import React from 'react';

import n from './NavSection.module.css';

const NavSection: React.FC = () => {
  return (
    <div className={n.AppWrapper}>
      <div className={n.weatherInfo}>
        <div className={n.icon}>
          <CloudOutlined width='30px' height='30px'/>
          <span>24Â°C</span>
        </div>
        <div className={n.desc}>
          <h4>SUNNY</h4>
          <span>sunny sunny today</span>
        </div>
      </div>

      <div className={n.foreCast}>
<div className={n.day}><span>oct</span><CloudOutlined /><span>14</span></div>
<div className={n.day}><span>oct</span><CloudOutlined /><span>14</span></div>
<div className={n.day}><span>oct</span><CloudOutlined /><span>14</span></div>
<div className={n.day}><span>oct</span><CloudOutlined /><span>14</span></div>
<div className={n.day}><span>oct</span><CloudOutlined /><span>14</span></div>
      </div>
      
<div className={n.sunset}>
<span>6:30</span>
<span>17:50</span>
    </div>
    <div className={n.group}>
<div className={n.something}>
  <h3>48</h3>
<span>something</span>
</div>
<div className={n.wind}>
  <h3>12</h3>
<span>m/s</span>
</div>
</div>
    </div>
  );
}

export default NavSection;