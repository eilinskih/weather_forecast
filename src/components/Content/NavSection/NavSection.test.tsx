
import { render } from '@testing-library/react'
import NavSection from './NavSection'

describe("navSection", () => {
    test("navSection snapshot", () => {
        const navSection = render(<NavSection weatherData={{main: "test", icon: "test", description: "test", temp: 1, sunrise: 1, sunset: 1, windSpeed: 1, clouds: 1, daily: [{ date: 1, icon: "test"
            }]}}/>)
            expect(navSection).toMatchSnapshot();
    })
})