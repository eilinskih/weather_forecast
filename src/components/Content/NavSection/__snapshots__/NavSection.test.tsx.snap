// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navSection navSection snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AppWrapper"
      >
        <div
          class="weatherInfo"
        >
          <div
            class="icon"
          >
            <img
              alt="img"
              src="http://openweathermap.org/img/wn/test@2x.png"
            />
            <span>
              1°C
            </span>
          </div>
          <div
            class="desc"
          >
            <h4>
              test
            </h4>
            <span>
              test
            </span>
          </div>
        </div>
        <div
          class="foreCast"
        >
          <div
            class="day"
          >
            <span>
              Oct
            </span>
            <img
              alt=""
              src="http://openweathermap.org/img/wn/test.png"
            />
            <span>
              20
            </span>
          </div>
        </div>
        <div
          class="sunset"
        >
          <span>
            sunrise
            <br />
            13: 3
          </span>
          <span>
            sunset
            <br />
            13: 3
          </span>
        </div>
        <div
          class="group"
        >
          <div
            class="something"
          >
            <h3>
              1%
            </h3>
            <span>
              clouds
            </span>
          </div>
          <div
            class="wind"
          >
            <h3>
              1 m/s
            </h3>
            <span>
              wind speed
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AppWrapper"
    >
      <div
        class="weatherInfo"
      >
        <div
          class="icon"
        >
          <img
            alt="img"
            src="http://openweathermap.org/img/wn/test@2x.png"
          />
          <span>
            1°C
          </span>
        </div>
        <div
          class="desc"
        >
          <h4>
            test
          </h4>
          <span>
            test
          </span>
        </div>
      </div>
      <div
        class="foreCast"
      >
        <div
          class="day"
        >
          <span>
            Oct
          </span>
          <img
            alt=""
            src="http://openweathermap.org/img/wn/test.png"
          />
          <span>
            20
          </span>
        </div>
      </div>
      <div
        class="sunset"
      >
        <span>
          sunrise
          <br />
          13: 3
        </span>
        <span>
          sunset
          <br />
          13: 3
        </span>
      </div>
      <div
        class="group"
      >
        <div
          class="something"
        >
          <h3>
            1%
          </h3>
          <span>
            clouds
          </span>
        </div>
        <div
          class="wind"
        >
          <h3>
            1 m/s
          </h3>
          <span>
            wind speed
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
